### PR TITLE
Create azure app service plugin nodeJsVersion that checks if nodeJs version is latest or not

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: plugins/azure/appservice/nodeJsVersion.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'NodeJs Version',
+    category: 'App Service',
+    domain: 'Application Integration',
+    severity: 'Low',
+    description: 'Ensures the latest version of NodeJs is installed for all App Services',
```
Comment (fatima99s) [inline plugins/azure/appservice/nodeJsVersion.js:9]:
```suggestion
    description: 'Ensures the latest version of Node.js is installed for all App Services',
```

File: plugins/azure/appservice/nodeJsVersion.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'NodeJs Version',
```
Comment (fatima99s) [inline plugins/azure/appservice/nodeJsVersion.js:5]:
```suggestion
    title: 'Node.js Version',
```

File: plugins/azure/appservice/nodeJsVersion.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    more_info: 'Installing the latest version of NodeJs will reduce the security risk of missing security patches.',
+    recommended_action: 'Select the latest version of NodeJs for all NodeJs-based App Services',
+    link: 'https://learn.microsoft.com/en-us/azure/app-service/configure-language-nodejs',
+    apis: ['webApps:list', 'webApps:listConfigurations'],
+    settings: {
+        latestNodeJsVersion: {
+            name: 'Latest NodeJs Version',
+            default: '20-lts',
+            description: 'The latest NodeJs version supported by Azure App Service.',
+            regex: '[0-9.]{1,2}-lts'
```
Comment (fatima99s) [inline plugins/azure/appservice/nodeJsVersion.js:19]:
```suggestion
            regex: '[0-9.]{1,2}'
```

Comment (fatima99s) [reply plugins/azure/appservice/nodeJsVersion.js:19]: handle the -lts logic in code

File: plugins/azure/appservice/nodeJsVersion.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            webApps.data.forEach(function(webApp) {
+                const webConfigs = helpers.addSource(
+                    cache, source, ['webApps', 'listConfigurations', location, webApp.id]
+                );
+
+                if (!webConfigs || webConfigs.err || !webConfigs.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query App Service: ' + helpers.addError(webConfigs),
+                        location, webApp.id);
+                } else {
```
Comment (fatima99s) [inline plugins/azure/appservice/nodeJsVersion.js:66]: add the logic for if the app is on windows platform

File: plugins/azure/appservice/nodeJsVersion.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    if (webConfigs.data[0] &&
+                        webConfigs.data[0].linuxFxVersion &&
+                        webConfigs.data[0].linuxFxVersion.indexOf('NODE') > -1 &&
+                        webConfigs.data[0].linuxFxVersion.indexOf('|') > -1) {
+                        found = true;
+                        console.log(webConfigs.data[0].linuxFxVersion);
```
Comment (fatima99s) [inline plugins/azure/appservice/nodeJsVersion.js:72]:
```suggestion
 
```

**Review by fatima99s — CHANGES_REQUESTED** [review]

File: plugins/azure/appservice/nodeJsVersion.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Node.js Version',
+    category: 'App Service',
+    domain: 'Application Integration',
+    severity: 'Low',
+    description: 'Ensures the latest version of Node.js is installed for all App Services',
+    more_info: 'Installing the latest version of Node.js will reduce the security risk of missing security patches.',
+    recommended_action: 'Select the latest version of Node.js for all Node.js-based App Services',
```
Comment (alphadev4) [inline plugins/azure/appservice/nodeJsVersion.js:11]:
```suggestion
    recommended_action: 'Select the latest version of Node for all Node.js based App Services',
```

File: plugins/azure/appservice/nodeJsVersion.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Node.js Version',
+    category: 'App Service',
+    domain: 'Application Integration',
+    severity: 'Low',
+    description: 'Ensures the latest version of Node.js is installed for all App Services',
+    more_info: 'Installing the latest version of Node.js will reduce the security risk of missing security patches.',
+    recommended_action: 'Select the latest version of Node for all Node.js based App Services',
```
Comment (alphadev4) [inline plugins/azure/appservice/nodeJsVersion.js:11]:
```suggestion
    recommended_action: 'Select the latest version of Node for all Node.js based App Services.',
```

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**